syntax = "proto3";

service BookInventory {

  rpc CreateBook (CreateBookRequest) returns (CreateBookResponse) {}

  rpc GetBook(GetBookRequest) returns (GetBookResponse) {}
}

enum GenreType {
  COMEDY = 0;
  THRILLER = 1;
  HORROR = 2;
  ROMANCE = 3;
}

enum StatusType {
  AVAILABLE = 0;
  TAKEN = 1;
}

message Book {
  optional string ISBN = 1;
  optional string title = 2;
  optional string author = 3;
  optional GenreType type = 4;
  optional int32 publishingYear = 5;
}

message InventoryItem {
  optional int32 InventoryNumber = 1;
  oneof Field {
    Book book = 2;
  }
  optional StatusType type = 3;
}

message GetBookRequest {
  string ISBN = 1;
}

message GetBookResponse {
  Book book = 1;
  string message = 2;
}

message CreateBookRequest {
  Book book = 1;
}

message CreateBookResponse {
  Book book = 1;
  string message = 2;
}